docs/modules/ROOT/pages/references/architecture/espejote-in-cluster-templating-controller.adoc

```diff
@@ -0,0 +1,502 @@
+
+    [ <16>
+      c {
+        metadata: {
+          name: 'copy-of-' + c.metadata.name,
+          // namespace: c.metadata.namespace, <17>
+          labels+: {
+            'espejote.io/created-by': 'copy-configmap'
+          }
+        }
+      },
+      for c in cm.items <18>
+    ]
+  serviceAccountName: my-service-account <19>
+----
+<1> `ManagedResource` are always namespace-scoped and can't access any resources outside of their namespace.
+<2> `applyOptions` can be used to resolve field conflicts.
+<3> `interval` triggers the template every 10 seconds.
+<4> `watchResource` triggers the template whenever the watched resource is created, changed, or deleted.
+<5> `name` is optional and can be used to select a specific resource.
+<6> `labelSelector` is optional and can be used to a specific set of resources.
+<7> `matchNames` is optional and can be used to filter the resources after the labelSelector is applied.
+<8> `ignoreNames` is optional and can be used to filter the resources after the labelSelector is applied.
+<9> `minTimeBetweenTriggers` is optional and can be used to limit the rate of re-renders.
+The re-renders are limited by the specific unique resource.
+<10> Context is a list of resources that are available in the Jsonnet template.
+They're not watched.
+<11> `def` defines a variable in the Jsonnet template.
+<12> `resource` is a Kubernetes resource that's available in the Jsonnet template.
+<13> `cache` is optional and can be used to setup a controller-runtime cache for the resource.
+The default is true.
+<14> `std.extVar` is used to access the trigger data.
+The full manifest is available if using the `watchResource` trigger.
+<15> `std.extVar` is used to access the context.
+It returns a object with all defined resources as keys.
+<16> The template can return a list of resources or a single resource.
+<17> The namespace of the resource will always be overwritten to the namespace of the `ManagedResource`.
+<18> Resource variables are always a list with zero or more items.
+
+
+=== `ClusterManagedResource`
+
+[source,yaml]
+----
+apiVersion: espejote.io/v1alpha1
+kind: ClusterManagedResource
+metadata:
+  name: default-network-policies
+spec:
+  triggers:
+  - watchResource:
+      apiVersion: v1
+      kind: Namespace <1>
+      labelSelector:
+        matchExpressions:
+        - key: inject-cm.syn.tools
+          operator: Exists
+  template: |
+    local cm = std.extVar('trigger').resource; <2>
+
+    [
+      {
+        metadata: {
+          name: 'injected-cm' + c.metadata.name,
+          namespace: c.metadata.namespace, <3>
+        }
+        stringData: {
+          'my-key': 'my-value',
+        }
+      }
+    ]
+----
+<1> Cluster scoped `ClusterManagedResource` can access all resources in the cluster.
+This includes cluster scoped resources.
+<2> The full manifest of the triggering is available.
+<3> A cluster managed resource can create resources in any namespace and thus needs to set the namespace.
```

Comment:
this is a bit inconsistent/incomplete, and I'm not sure what the actual example wants to do. I'd like to have a consistent example here (either copy config map to target namespace -- in that case the source is missing, or generating netpols, in which case the full object should be visible in Jsonnet?